geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,47 +238,89 @@ object IngestCommand extends LazyLogging {
 
     var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
 
+    // if there is no converter passed in
     if (converter == null && inputs.nonEmpty) {
-      // if there is no converter passed in, try to infer the schema from the input files themselves
-      Command.user.info("No converter defined - will attempt to detect schema from input files")
-      val file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
-        throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+      var open: () => InputStream = null
+      var file: FileHandle = null
+
+      // try to infer the schema from stdin
+      if (inputs == Inputs.StdInInputs) {
+        Command.user.info("No converter defined - will attempt to detect schema from stdin")
+
+        // Cache the bytes from the input stream, because we'll need them again for the ingest
+        val inputBytes = System.in.readAllBytes()
```

Comment:
I'm a little wary about reading in all available bytes, as it might blow out memory. All the processing during ingest is streaming, so we don't generally hold much data in memory. i think ideally we could read in some fixed amount of bytes, run type inference on that, then for the actual ingest create an input stream that first returns the cached bytes, then starts reading from stdin again.

Reply:
That makes sense, but how would we determine the number of bytes to read in and do type inference on?

In order to infer the SFT, I think we'd need to read in one SimpleFeature from stdin (at minimum), but we wouldn't know the size of that SimpleFeature at compile time.

Reply:
`AbstractConverterFactory.inferSampleSize` provides a configuration for how many lines should be considered in type inference. 

I would suggest something like defining an input iterator near where your `var open` is. You can duplicate the input stream and sample one duplicate for use in the inferencer and save one for ingest later. 
```
val (input, sample) = Iterator.continually(io.StdIn.readLine()).duplicate match { 
  case (i, s) => (i, s.take(AbstractConverterFactory.inferSampleSize).toList)
}
```

Reply:
yeah, delegating to inferSampleSize would be ideal since it's already there and documented. you might also be able to create a new wrapper class that implements `() => InputStream` (which is the factory method used by the type inference) that could handle caching the bytes as they get read, then re-setting back to return the cached bytes.